packages/core/src/server.ts

```diff
@@ -646,11 +646,13 @@ export class Controller extends API {
         authToken,
         verify,
     }: CreateAccountParams): Promise<Account> {
-        // For compatibility with v3 clients, which still use the depracated `verify` property name
+        // For compatibility with v3 clients, which still use the deprecated `verify` property name
         if (verify && !authToken) {
             authToken = verify;
         }
 
+        account.validateOrThrowInputValues();
```

Comment:
This will take in consideration the server-side validation, which only happens after the email, name, and master password have been set, whereas the client-side happens right on field submission/keyup.